src/main/java/control/ToolViewController.java

```diff
@@ -93,26 +97,41 @@ private void showBlockCreationWindow(MouseEvent event) {
     }
 
     /**
-     * Event handler for the creation of a director shot.
-     * It adds the DirectorShot to the DirectorTimeline and adds the corresponding
+     * When triggered, this initializes and displays the modal view for the creation of
+     * a new DirectorBlock.
+     * @param event mouse event
+     */
+    private void showDirectorBlockCreationWindow(MouseEvent event) {
+        new DirectorShotCreationModalView(this.controllerManager.getRootPane(),
+            this.controllerManager.getScriptingProject().getCameraTimelines().size(),
+            this::createDirectorShot);
+    }
+
+    /**
+     * Event handler for the creation of a camera shot.
+     * It adds the CameraShot to the CameraTimeline and adds the corresponding
      * camera shots via the TimelineController.
      * @param event shot creation event
      */
-    private void createDirectorShot(DirectorShotCreationEvent event) {
-        ScriptingProject script = this.controllerManager.getScriptingProject();
-
-        script.getDirectorTimeline().addShot(event.getShotName(), event.getShotDescription(),
-                                             event.getShotStart(), event.getShotEnd());
-
-        TimelineController timelineController = this.controllerManager.getTimelineControl();
+    private void createCameraShot(CameraShotCreationEvent event) {
+        TimelineController tlineControl = this.controllerManager.getTimelineControl();
         event.getCamerasInShot().forEach(camInd -> {
-                timelineController.addCameraShot(camInd, event.getShotName(),
-                        event.getShotDescription(), event.getShotStart(),
-                        event.getShotEnd());
+                tlineControl.addCameraShot(camInd,
```

Comment:
This creates the camera shots in the checked timelines. This is not the create method for the DirectorShot.